File: stack-using-queue/StackQueue.cs
using System.Collections.Generic;

public class StackQueue {

    private List<int> list = new List<int>();

    public MyStack() {}
    
    public void Push(int x) {
        this.list.Add(x);
    }
    
    public int Pop() {
        int index = this.list.Count - 1;
        int item = this.list[index];
        this.list.RemoveAt(index);
        return item;
    }
    
    public int Top() {
        return this.list[list.Count - 1];
    }
    
    public bool Empty() {
        return this.list.Count == 0;
    }
}

File: stack-using-queue/StackQueue.js
var StackQueue = function() {
    this.stack = [];
};

/** 
 * @param {number} x
 * @return {void}
 */
StackQueue.prototype.push = function(x) {
    this.stack.push(x);
};

/**
 * @return {number}
 */
StackQueue.prototype.pop = function() {
    return this.stack.pop();
};

/**
 * @return {number}
 */
StackQueue.prototype.top = function() {
    return this.stack[this.stack.length - 1];
};

/**
 * @return {boolean}
 */
StackQueue.prototype.empty = function() {
    return this.stack.length === 0;
};